src/util.ts

```diff
@@ -338,3 +338,49 @@ export function getValueFromHeaders(headers: IncomingHttpHeaders, key: string):
 export function mapToObject(map: Map<any, any>): object {
 	return Object.fromEntries(Array.from(map.entries(), ([k, v]) => v instanceof Map ? [k, mapToObject(v)] : [k, v]));
 }
+
+export function isValidBirthday(dateString: string): boolean {
+	// * Birthdays MUST be in the format YYYY-MM-DD. This is how the
+	// * console sends them, regardless of region
+
+	// * Make sure general format is right
+	const regex = /^\d{4}-\d{2}-\d{2}$/;
+	if (!regex.test(dateString)) {
+		return false;
+	}
+
+	// * Actually check that it's a valid date
+	const parts = dateString.split('-');
+	const year = parseInt(parts[0], 10);
+	const month = parseInt(parts[1], 10);
+	const day = parseInt(parts[2], 10);
+
+	const date = new Date(year, month - 1, day);
+
+	return date.getFullYear() === year && date.getMonth() === month - 1 && date.getDate() === day;
+}
+
+export function getAgeFromDate(dateString: string): number {
+	if (!isValidBirthday(dateString)) {
+		return -1;
+	}
+
+	const parts = dateString.split('-');
+	const birthYear = parseInt(parts[0], 10);
+	const birthMonth = parseInt(parts[1], 10);
+	const birthDay = parseInt(parts[2], 10);
+
+	const today = new Date();
+	const currentYear = today.getFullYear();
+	const currentMonth = today.getMonth() + 1;
+	const currentDay = today.getDate();
+
+	let age = currentYear - birthYear;
+
+	// * Check if birthday has actually happened this year yet
+	if (currentMonth < birthMonth || (currentMonth === birthMonth && currentDay < birthDay)) {
+		age--;
+	}

```

Comment:
I think it might be better to do this with maths

```js
const yearMs = 1000 * 60 * 60 * 24 * 365;
const dateDiff = new Date() - new Date(currentYear, currentMonth, currentDay);
const age = Math.floor(dateDiff / yearMs);
```